tests/LinkTest.php

```diff
@@ -6,10 +6,76 @@ class LinkTest extends CopilotTagTest
 {
     public function expectedWrites()
     {
+        $assetMarkdown = "\n\n[#image: /photos/123ID]|||some caption|||\n";
+
         return [
             [
-                new Link("Hello world!"),
-                "Hello world!"
+                new Link(),
+                ""
+            ],
+            [
+                new Link(""),
+                ""
+            ],
+            [
+                new Link("Hello world!", "http://li.nk"),
+                "[Hello world!](http://li.nk)"
+            ],
+            [
+                new Link("Hello world!", "http://li.nk", Array("target" => "_blank")),
+                "[Hello world!](http://li.nk){: target=\"_blank\" }"
+            ],
+            [
+                new Link("Hello world!", "http://li.nk", Array("target" => "_blank", "title" => "rawr")),
+                "[Hello world!](http://li.nk){: target=\"_blank\" title=\"rawr\" }"
+            ],
+            [
+                new Link("Hello world!", null, Array("target" => "_blank", "title" => "rawr")),
+                "[Hello world!](){: target=\"_blank\" title=\"rawr\" }"
+            ],
+            [
+                new Link("Hello world!\n\n", "http://li.nk"),
+                "[Hello world!](http://li.nk)\n\n"
+            ],
+            [
+                new Link("  Hello world!\n\n", "http://li.nk"),
+                "  [Hello world!](http://li.nk)\n\n"
+            ],
+            [
+                new Link("  Hello world!\n\n", "http://li.nk"),
+                "  [Hello world!](http://li.nk)\n\n"
+            ],
+            [
+                new Link("Hello world!", "http://li.nk     "),
+                "[Hello world!](http://li.nk)"
+            ],
+            [
+                new Link($assetMarkdown, "http://li.nk"),
+                "$assetMarkdown"
+            ],
+            [
+                new Link("Hello\nworld!", "http://li.nk"),
+                "[Hello](http://li.nk)\n[world!](http://li.nk)"
+            ],
+            [
+                new Link("{$assetMarkdown}{$assetMarkdown}", "http://li.nk"),
+                "{$assetMarkdown}{$assetMarkdown}"
```

Comment:
Am I correct in thinking would be 3 newlines in the middle of this string? We should be reducing 3 newlines to 2 in the beautify function I thought?